### ISSUE TITLE
set option <30120> failed: <A libcurl function was given a bad argument>

### ISSUE THREAD

**Issue by cnbleu** [open-telemetry/opentelemetry-cpp#2175]
Hey all, 

There is an error message `ext\src\http\client\curl\http_operation_curl.cc:548 CURL, set option <30120> failed: <A libcurl function was given a bad argument>` while running the `example_otlp_http` example with Win32 on my Windows 10. I have already debug it and found that after add these follow codes, it works well.

```
    #ifdef WIN32
    rc = SetCurlLongOption(CURLOPT_POSTFIELDSIZE, req_size);
    #else
    rc = SetCurlLongOption(CURLOPT_POSTFIELDSIZE_LARGE, req_size);
    #endif
```

It should be a 32-bit compatibility problem.

**Comment by owent** [comment]
Maybe we can always use `CURLOPT_POSTFIELDSIZE_LARGE` .

**Comment by cnbleu** [comment]
Hi, ownet. Do you mean we can always use `CURLOPT_POSTFIELDSIZE_LARGE `? This could be problematic. It not works on Win32, it will print `548 CURL, set option <30120> failed: <A libcurl function was given a bad argument>`. We should use 
 `CURLOPT_POSTFIELDSIZE` on Win32.

**Comment by owent** [comment]
> Hi, ownet. Do you mean we can always use `CURLOPT_POSTFIELDSIZE_LARGE `? This could be problematic. It not works on Win32, it will print `548 CURL, set option <30120> failed: <A libcurl function was given a bad argument>`. We should use `CURLOPT_POSTFIELDSIZE` on Win32.

Maybe it's better to add a `SetCurlOffOption` and take `curl_off_t` as value. According to https://curl.se/libcurl/c/CURLOPT_POSTFIELDSIZE_LARGE.html, `CURLOPT_POSTFIELDSIZE_LARGE` use `curl_off_t` . And there also are some other options use this type.
The `long` type is different on different platforms and environments, and we should let compiler to decide whether it's the same as `curl_off_t`.

**Comment by marcalff** [comment]
This looks like a regression introduced by:
* https://github.com/open-telemetry/opentelemetry-cpp/pull/1793

because the helper `SetCurlLongOption()` was used on a type curl_off_t.

We need a `SetCurlOffOption()` indeed.

**Comment by marcalff** [comment]
@cnbleu Thanks for the report.

Could you confirm if this fix works for you:
* https://github.com/open-telemetry/opentelemetry-cpp/pull/2178

Thanks.

**Comment by cnbleu** [comment]
> @cnbleu Thanks for the report.
> 
> Could you confirm if this fix works for you:
> 
> * [[EXPORTER] Fixed HTTP CURL for 32bits platforms #2178](https://github.com/open-telemetry/opentelemetry-cpp/pull/2178)
> 
> Thanks.

Hi @marcalff , i have re-compile with your branch and test it on my Windows, it works. Thank you.

**Comment by Korevishal** [comment]
How to use this change? is it available with 1.9.1 release ?
I believe I have to sync the main and build the Openetelemetry binaries, do I need to call SetCurlOffOption explicitly from my client ? if yes what values would I need to pass.
I am also seeing this issue on 32 bit windows
